packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -574,7 +567,6 @@ class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTr
       curve: const Interval(0.0, 0.1),
     );
 
-    _previousRotationAnimation = new TrainHoppingAnimation(previousExitRotationAnimation, moveRotationAnimation);
```

Comment:
i guess one case where the behaviour would change here is the case where moveRotationAnimation starts, or passes through, 1.0. Then, the _previousRotationAnimation would hop to the moveRotationAnimation. Is that intentional?

I don't really understand what this code is trying to do.

Reply:
That is what happens with the bottom app bar notch demo.

When the FAB is moving and exiting, the user-defined moveRotationAnimation could cause the scale animation to go past 1.0, but both animations are running.

With the default specification, this only affects the current rotation animation, because the default moveRotationAnimation doesn't rotate on the way out.  However, it is customizable, so a value that does this is possible.

When writing this, we didn't have a good way of testing this -- the controller is private, so the test would need to either expose the controller for testing purposes or depend on the internal implementation of the RotationTransition.

My thinking is about the first is that it would be better to avoid exposing private members in our public API just to test them.

My thinking about the second is that it would be hard to debug if a change to the RotationTransition's implementation were to break this.

Since this isn't clear, I think the best resolution is to comment, explaining the purpose of this code.  I can summarize this discussion into a comment on the code..

Reply:
Well my real concern is that `previousExitRotationAnimation` seems like an expensive way of doing nothing. If all we want is to catch when `moveRotationAnimation` hits 1.0, then maybe an `AlwaysStoppedAnimation` would be better than the current setup?

The other concern is that I deleted all this code and no test failed.

Reply:
@DaveShuckerow Would you be willing to add a test to verify the behaviour that I'd be breaking here? Then I can see if there's a better way to refactor the code. At the moment I'm a bit scared of touching this code since I stripped all this code and no test broke.

Reply:
ping @DaveShuckerow